### ISSUE TITLE
no-unbound-method rule doesn't work with destructuring

### ISSUE THREAD

**Issue by karol-majewski** [palantir/tslint#3884]
### Bug Report

- __TSLint version__: 5.10.0
- __TypeScript version__: 2.8.3
- __Running TSLint via__: CLI

#### TypeScript code being linted

```ts
const { stringify } = JSON;
```

with `tslint.json` configuration:

```json
{
  "extends": [
    "tslint:latest"
  ],
  "rules": {
    "no-unbound-method": true
  }
}

```

#### Actual behavior

No error.

#### Expected behavior

Error: `Avoid referencing unbound methods which may cause unintentional scoping of 'this'.`

`const { stringify } = JSON;` is equivalent to `const stringifyFn = JSON.stringify;`, yet only the latter triggers the warning.

**Comment by JoshuaKGoldberg** [comment]
### 💀 _**It's time!**_ 💀

**TSLint is deprecated** and no longer accepting pull requests other than security fixes. See #4534. ☠️ 
We recommend you instead use **[typescript-eslint](https://typescript-eslint.io)** to lint your TypeScript code with **ESLint**. ✅

👋 It was a pleasure open sourcing with you!

**Comment by JoshuaKGoldberg** [comment]
🤖 Beep boop! 👉  **TSLint is deprecated** 👈 _(#4534)_  and you should **switch to [typescript-eslint](https://typescript-eslint.io)**! 🤖 

🔒 This issue is being locked to prevent further unnecessary discussions. Thank you! 👋
